### PR TITLE
Engine cod

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 19 functions (+0.30%) on `us`.

⚠️ This PR introduces 6 warnings:
- src/163400.c:107: warning: initialization makes integer from pointer without a cast
- src/163400.c:75: warning: initialization makes integer from pointer without a cast
- src/163400.c:91: warning: initialization makes integer from pointer without a cast
- src/182B30.c:908: warning: passing arg 4 of `spr_draw_player_sprite' from incompatible pointer type
- src/182B30.c:911: warning: passing arg 4 of `spr_draw_player_sprite' from incompatible pointer type
- src/a5dd0_len_114e0.c:1362: warning: passing arg 2 of `bind_entity_model_setupGfx' makes pointer from integer without a cast
- src/a5dd0_len_114e0.c:1367: warning: passing arg 2 of `bind_entity_model_setupGfx' makes pointer from integer without a cast
- src/a5dd0_len_114e0.c:1417: warning: passing arg 2 of `bind_entity_model_setupGfx' makes pointer from integer without a cast
- src/status_icons.c:204: warning: assignment from incompatible pointer type
- src/status_icons.c:205: warning: assignment from incompatible pointer type
- src/status_icons.c:287: warning: assignment from incompatible pointer type
- src/status_icons.c:290: warning: assignment from incompatible pointer type

File: include/controller.h [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+#include "PR/os_internal.h"
+#include "PR/rcp.h"
+
+//should go somewhere else but
+#define ARRLEN(x) ((s32)(sizeof(x) / sizeof(x[0])))
+#define CHNL_ERR(format) (((format).rxsize & CHNL_ERR_MASK) >> 4)
+
+typedef struct
+{
```
Comment (nanaian) [inline include/controller.h:12]: This file has weird formatting -- has it come from another project?

File: src/101b90_len_8f0.c [hunk 2]
```diff
@@ -1,19 +1,57 @@
...
+
+    nuPiReadRom(SPRITE_ROM_START, &D_802E0C60, 0xC);
+    D_802E0C58 = cacheSize;
+    D_802E0C5C = maxRasterSize;
+    D_802E0C60[0] += SPRITE_ROM_START;
+    D_802E0C60[1] += SPRITE_ROM_START;
+    D_802E0C60[2] += SPRITE_ROM_START;
+    var_a1 = _heap_malloc(&gSpriteHeapPtr, maxRasterSize * cacheSize);
+
+    for (i = 0; i < 0x12; i++) {
```
Comment (nanaian) [inline src/101b90_len_8f0.c:40]: `ARRAY_LEN(D_802E0C70)`?

File: include/controller.h (hunk 1 above)
Comment (ethteck) [reply include/controller.h:12]: Yeah, ultralib. I'd prefer to keep it for now since we'll eventually probably subrepo that anyway

**Pull request merged**